sites/default/statement.inc.php

```diff
@@ -21,13 +22,373 @@
+ *  @param string $pid patient_id
+ *  @param string $direction, options "web" or anything else.  Web provides apache-friendly url links.
+ *  @return outputs to be displayed however requested
+ */
+function report_header_2($stmt,$direction='',$providerID='1') {
+  $titleres = getPatientData($stmt['pid'], "fname,lname,DOB");
+  if ($_SESSION['pc_facility']) {
+    $sql = "select * from facility where id=?";
+    $facility = sqlQuery($sql,array($_SESSION['pc_facility']));
+  } else {
+    $sql = "SELECT * FROM facility ORDER BY billing_location DESC LIMIT 1";
+    $facility = sqlQuery($sql);
+  }
+  $DOB = oeFormatShortDate($titleres['DOB']);
+  /******************************************************************/
+  ob_start();
+  // Use logo if it exists as 'practice_logo.gif' in the site dir
+  // old code used the global custom dir which is no longer a valid
+  ?>
+  <table style="width:7in;">
+    <tr>
+      <td style='width:100px;text-align:top;'>
+        <?php
+          $practice_logo = $GLOBALS['OE_SITE_DIR']."/images/practice_logo.gif";
+          if (file_exists($practice_logo)) {
+            echo "<img src='$practice_logo' align='left' style='width:125px;margin:0px;'><br />\n";
+          }
+        ?>
+      </td>
+      <td style='width:40%;'>
+        <em style="font-weight:bold;font-size:1.4em;"><?php echo text($facility['name']); ?></em><br />
+        <?php echo text($facility['street']); ?><br />
+        <?php echo text($facility['city']); ?>, <?php echo text($facility['state']); ?> <?php echo text($facility['postal_code']); ?><br />
+        <?php echo xlt('Phone').': ' .text($facility['phone']); ?><br />
+        <?php echo xlt('Fax').': ' .text($facility['fax']); ?><br />
+        <br clear='all' />
+      </td>
+      <td>
+        <em style="font-weight:bold;font-size:1.4em;"><?php echo text($titleres['fname']) . " " . text($titleres['lname']); ?></em><br />
+        <b style="font-weight:bold;"><?php echo xlt('Chart Number'); ?>:</b> <?php echo text($stmt['pid']); ?><br />
+        <b style="font-weight:bold;"><?php echo xlt('Generated on'); ?>:</b> <?php echo oeFormatShortDate(); ?><br />
+        <b><?php echo xlt('Provider') . ':</b>  '; ?><?php echo text(getProviderName($providerID)); ?> <br />
+      </td>
+    </tr>
+  </table>
+  <?php
+  $output = ob_get_contents();
+  ob_end_clean();
+  return $output;
+}
+
+function create_HTML_statement($stmt) {
+  if (! $stmt['pid']) return ""; // get out if no data
+
+  #minimum_amount_due_to _print
+  if ($stmt['amount'] <= ($GLOBALS['minimum_amount_to_print']) && $GLOBALS['use_statement_print_exclusion']) return "";
+
+  // Facility (service location)
+  $atres = sqlStatement("select f.name,f.street,f.city,f.state,f.postal_code,f.attn,f.phone from facility f " .
+    " left join users u on f.id=u.facility_id " .
+    " left join  billing b on b.provider_id=u.id and b.pid = '".add_escape_custom($stmt['pid']). "' ".
+    " where  service_location=1");
```

Comment:
Just give it a go. It should work.(note your prior commit had the array in the below sqlFetchArray which is why it didn't work). Definitely should be able to escape variables within join. For example:
https://github.com/openemr/openemr/blob/ff2313cb47e05d3a33812d6346d6bc801fcf7fc8/myportal/soap_service/server_newpatienttoapprove.php#L395-L396
(if it doesn't work on a quick test, then definitely ok to leave it escaped with add_escape_custom though)